cmd/kube-proxy/app/server.go

```diff
@@ -620,7 +621,13 @@ func (s *ProxyServer) Run(ctx context.Context) error {
 		HealthServer: s.HealthzServer,
 	})
 	nodeConfig.RegisterEventHandler(s.Proxier)
-
+	// TODO: remove once ConsistentReadFromCache and/or WatchList graduate to GA
+	// Informers may get stale data, specially in cases where the kube-proxy runs as a static pod
+	// or an independent binary, since it may run before the kubelet on the node updates the Node.
+	// The solution in the meantime is to process the Node events and crash if the NodeIPs
+	// has changed since the first read.
+	// https://issues.k8s.io/111321
```

Comment:
This isn't about "stale data" in that sense; it's not that kube-proxy is seeing incorrect data, it's that it's possible for the data to change after kube-proxy first learns it (if the node is deleted and then recreated). Those features wouldn't help.